notation.go

```diff
@@ -66,6 +62,10 @@ type signerAnnotation interface {
 type SignOptions struct {
 	SignerSignOptions
 
+	// ArtifactReference sets the reference of the artifact that needs to be
+	// signed.
```

Comment:
We need to document that `ArtifactReference` is a tag or a digest or a full reference.

Similarly, we need to document `ArtifactReference` in `VerifierVerifyOptions` that it must be a full reference.